hazelcast-jet-hadoop/src/main/java/com/hazelcast/jet/hadoop/impl/ReadHdfsNewApiP.java

```diff
@@ -150,7 +153,21 @@ public boolean complete() {
 
     private static InputFormat getInputFormat(Configuration configuration) {
         Class<?> inputFormatClass = configuration.getClass(MRJobConfig.INPUT_FORMAT_CLASS_ATTR, TextInputFormat.class);
-        return (InputFormat) ReflectionUtils.newInstance(inputFormatClass, configuration);
+        try {
+            return (InputFormat) ReflectionUtils.newInstance(inputFormatClass, configuration);
+        } finally {
+            removeInputFormatFromCache(inputFormatClass);
+        }
+    }
+
+    private static void removeInputFormatFromCache(Class<?> inputFormatClass) {
```

Comment:
May be it is better to clear the cache rather than removing single class from there in the close method ? WDYT?

Reply:
since it is a static field, I didn't want to touch more than what we put there. maybe other parts of hadoop uses this util internally